### PR TITLE
Allow handler-like objects within routing

### LINKED ISSUE #159: Routes that were working on 0.5.5 not working on 0.6.2
Greetings, thanks for the project. 

I am having issues after upgrading where it seems one of my routes is getting converted to a coroutine somewhere and not awaited and then is being passed down all the middleware.

```
Traceback (most recent call last):
  File "../lib/python3.7/site-packages/uvicorn/protocols/http/httptools_impl.py", line 387, in run_asgi
    result = await asgi(self.receive, self.send)
  File "../lib/python3.7/site-packages/uvicorn/middleware/message_logger.py", line 59, in __call__
    await self.inner(self.receive, self.send)
  File "../lib/python3.7/site-packages/uvicorn/middleware/debug.py", line 80, in __call__
    await asgi(receive, self.send)
  File "../lib/python3.7/site-packages/starlette/exceptions.py", line 69, in app
    raise exc from None
  File "../lib/python3.7/site-packages/starlette/exceptions.py", line 61, in app
    await instance(receive, sender)
TypeError: 'coroutine' object is not callable
```

### PR DESCRIPTION
**Pull request by Dith3r**
Small change to allow to use class methods or object methods as proper endpoints for Routes.

Example:

```python

test_uuid = "(?P<test_uuid>[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})"

app = Starlette()


class TestHandler:
    def __init__(self):
        # dependencies for handling request
        pass

    async def get(self, request: Request):
        return UJSONResponse(request.path_params)


test_handler = TestHandler()


app.add_route(f'/test/{test_uuid}', test_handler.get)
```

Closes #159

### CODE REVIEW

File: starlette/routing.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def __init__(
         self.endpoint = endpoint
         self.name = get_name(endpoint)
 
-        if inspect.isfunction(endpoint):
+        if inspect.isfunction(endpoint) or inspect.ismethod(endpoint):
```
Comment (lovelydinosaur) [inline starlette/routing.py:87]: Yup, looks good to me. Let's use the same for `WebSocketRoute` too.

**Comment by lovelydinosaur** [comment]
Thanks!

**Comment by jared-mackey** [comment]
Perfect, thanks!

**Comment by lovelydinosaur** [comment]
Now released in 0.6.3. https://www.starlette.io/release-notes/

**Pull request merged**